umbraco/Umbraco-CMS
# Allow to pass in boolean to noDirtyCheck directive

### Prerequisites


### Description
At the moment we have some duplicated logic to be able to set `no-dirty-check` directive on e.g. `<umb-checkbox>` and `<umb-radiobutton>`.

I would however like just to pass in a boolean value whether to set this or not.

For example:

- `no-dirty-check` without attribute value set `$dirty` to false.
- `no-dirty-check="true"` with attribute value `true` works as before mentioned.
- `no-dirty-check="false"` with attribute value `true` works as default behaviour.

The directive was originally created based on this https://stackoverflow.com/a/27430980 but note sure why we need the local variable with getter and setter. Furthermore why it change `$pristine`. It seems this was changed in https://github.com/umbraco/Umbraco-CMS/pull/2476

I found some different variants setting `$dirty` to `false` or `angular.noop`:
https://stackoverflow.com/a/29225603


![2020-08-17_14-07-38](https://user-images.githubusercontent.com/2919859/90394582-34456980-e093-11ea-922a-f7783d82b690.gif)


### Test notes

- Check that changes to input fields in content trigger the "Unsaved changes" dialog
- Add a custom dashboard with use on `no-dirty-check`,  `no-dirty-check="true"` and `no-dirty-check="true"` on `<input>` field works as expected. Also check with`<umb-checkbox>` and `<umb-radiobutton>` using `disable-dirty-check` property with `true/false` value.


```
<div>
    <label>Text 1</label>:<br>
    <input type="text" name="text1" placeholder="Enter value..." ng-model="text1" no-dirty-check />
</div>

<div>
    <label>Text 2</label>:<br>
    <input type="text" name="text2" placeholder="Enter value..." ng-model="text2" no-dirty-check="false" />
</div>

<div>
    <label>Text 3</label>:<br>
    <input type="text" name="text3" placeholder="Enter value..." ng-model="text3" no-dirty-check="true" />
</div>

<div>
    <label>Checkbox</label>:<br>
    <umb-checkbox model="check1" name="check1" disable-dirty-check="false">Test Check</umb-checkbox>
</div>

<div>
    <label>Radiobutton</label>:<br>
    <umb-radiobutton model="radio1" name="radio1" disable-dirty-check="true">Test Radio</umb-radiobutton>
</div>
```

[MyDashboard.zip](https://github.com/umbraco/Umbraco-CMS/files/5084038/MyDashboard.zip)
